gcc/rust/backend/rust-compile-intrinsic.cc

```diff
@@ -82,11 +101,128 @@ Intrinsics::compile (TyTy::FnType *fntype)
   if (builtin != nullptr)
     return builtin;
 
+  // is it an generic builtin?
+  for (size_t i = 0; i < GENERIC_INTRINSIC_SIZE; i++)
+    {
+      const BuiltinGenericIntrinsic &intrinsic = generic_intrinsics[i];
+      bool found
+	= intrinsic.identifier.compare (fntype->get_identifier ()) == 0;
+      if (found)
+	return intrinsic.handler (ctx, fntype);
+    }
```

Comment:
```suggestion
// is it a generic builtin?
auto it = generic_intrinsics_map.find (fntype->get_identifier ());

if (it != generic_intrinsics_map.end ())
   return it->second (ctx, fntype);
```

Reply:
yeah this sounds good i forget you can statically initialize a std::map

Reply:
fixed